### DISCUSSION TITLE
How does the Quasar bootstrap process work?

### DISCUSSION THREAD

**Discussion by kev-reth-va** [quasarframework/quasar#14357] [General - Components / Directives / etc]
In a Vue app, the starting point index.html calls a script which does something like this

```
import App from "../../App.vue";
const app = createApp(App);
app.mount("#app");
```

In the default Quasar CLI install I see nothing like this. Instead index.html has a comment which apparently bootstraps Quasar and Vue:
```
  <body>
    <!-- quasar:entry-point -->
  </body>
```

Beyond that, index.html can be relatively empty. Although similar behavior may be available to Vue, I haven't seen it.

How does the browser initialize Vue and Quasar from index.html with only a comment as a reference? I would think there would need to be at least a required script. The browser would normally skip `<!-- quasar:entry-point -->`.

**Comment by lucianobosco** [comment]
Not sure if I understand your question, but Quasar uses Boot files to run code before the root Vue app instance is instantiated.
I assume that Quasar has internal boot files for registering and mounting the entry point (AKA main.js in a pure Vue CLI app).
You can also use your own Boot files to add any logic you want before app mounts (such as authentication, global components, i18n and other plugins settings)
https://quasar.dev/quasar-cli-vite/boot-files#introduction

**Reply by kev-reth-va** [reply]
> Not sure if I understand your question, but Quasar uses Boot files to run code before the root Vue app instance is instantiated. I assume that Quasar has internal boot files for registering and mounting the entry point (AKA main.js in a pure Vue CLI app). You can also use your own Boot files to add any logic you want before app mounts (such as authentication, global components, i18n and other plugins settings) https://quasar.dev/quasar-cli-vite/boot-files#introduction

Thanks. How do those boot files get loaded? Normally the browser will skip the commented directive `<!-- quasar:entry-point -->` Obviously it somehow gets executed but without any scripts in index.html I'm not sure how. The browser would just display a blank page. I don't see what prompts the browser's Javascript engine to initialize Quasar.

I'll look into boot files, thanks for the information.

=== Addendum
To further clarify, this is my index.html, which of course is the entry point to the application. I don't see why it displays anything other than a blank page, but it does indeed start Quasar.

```
<!DOCTYPE html>
<html>
  <head>
    <title></title>
  </head>
  <body>
    <!-- quasar:entry-point -->
  </body>
</html>
```

**Comment by metalsadman** [comment]
Check the files inside the generated .quasar folder, should be able to get the idea how the quasar app starts.

**Reply by kev-reth-va** [reply]
> Check the files inside the generated .quasar folder, should be able to get the idea how the quasar app starts.

Thanks. `.quasar/client-entry.js` appears to be where quasar starts executing. I don't understand is how this happens without something like  `<script src="myscripts.js"></script>` in `index.html`. There's just `<!-- quasar:entry-point -->` which the browser would ordinarily ignore and display a blank page. How does the browser's Javascript engine knows about the Quasar boot files? There's no reference to them in `index.html`.

I think I realized what's going on here. Starting the `quasar dev` server makes this magic happen. I was thinking as if the browser looks for index.html, but it's actually the server that does so. Apparently the Quasar server processes index.html to do everything required.
